### PR TITLE
PHPLIB-104: Operation classes for some commands

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/PHPLIB-104

### CODE REVIEW

**Comment by bjori** [comment]
[That's hot!](http://i.giphy.com/XbqnnugDSyNby.gif)

File: src/Operation/Aggregate.php [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        }
+
+        return new ArrayIterator(array_map(
+            function (\stdClass $document) { return (array) $document; },
+            $result['result']
+        ));
+    }
+
+    /**
+     * @param boolean $isCursorSupported
```
Comment (bjori) [inline src/Operation/Aggregate.php]: ?

Comment (jmikola) [inline src/Operation/Aggregate.php]: This was leftover from when the wire version check was in `execute()` (before I moved it to `createCommand()`).

File: src/Operation/Aggregate.php [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+/**
+ * Operation for the aggregate command.
+ *
+ * @api
+ * @see MongoDB\Collection::aggregate()
+ * @see http://docs.mongodb.org/manual/reference/command/aggregate/
+ */
+class Aggregate implements Executable
+{
+    private static $wireVersionForCursor = 2;
```
Comment (jmikola) [inline src/Operation/Aggregate.php]: Before you ask, this is the closest thing to a private constant. I'd rather not have this in the public API.

**Comment by bjori** [comment]
lgtm

File: tests/FunctionalTestCase.php [hunk 3]
```diff
@@ -42,6 +46,7 @@ protected function getServerVersion(ReadPreference $readPreference = null)
             $readPreference ?: new ReadPreference(ReadPreference::RP_PRIMARY)
         );
 
+        $cursor->setTypeMap(array('document' => 'array'));
```
Comment (jmikola) [inline tests/FunctionalTestCase.php]: @derickr: This should address the error you were seeing.

**Pull request merged**